features/step_definitions/us20_borrower_views_deed.rb

```diff
@@ -2,11 +2,19 @@
   visit(Env.borrower_frontend + path)
 end
 
-When(/^I search for the created deed$/) do
-  url = JSON[@response.body]["url"]
-  split_url = url.split('/')
-  @deed_id = split_url[4]
-  fill_in 'deed_reference', with: @deed_id
+When(/^I retrieve the deed id$/) do
+  response_hash = JSON.parse(@response.body)
+  @get_url = response_hash['url']
+end
+
+And(/^I retrieve the unique user id using the URL$/) do
+  @response = HTTP.get(@get_url)
+  @deed_hash = JSON.parse(@response.body)
```

Comment:
This deed_hash variable should probably be local as it's only used within this step, so remove the @ from the beginning of it.